genesyscloud/util_retries.go

```diff
@@ -19,7 +19,8 @@ import (
 func withRetries(ctx context.Context, timeout time.Duration, method func() *resource.RetryError) diag.Diagnostics {
 	err := diag.FromErr(resource.RetryContext(ctx, timeout, method))
 	if err != nil && strings.Contains(fmt.Sprintf("%v", err), "timeout while waiting for state to become") {
-		ctx, _ := context.WithTimeout(context.Background(), timeout)
```

Comment:
We should never ignore the cancel object and should explicitly do a defer cancel() on the returned function.  This was caught by go vet.